lib/storage.js

```diff
@@ -67,24 +68,60 @@ async function putRendition(rendition) {
         // asset-compute-cli command run-worker wants file named as originally requested
         // however, computing the metadata wants the current path, so
         // we need to make a copy not just rename
-        if(rendition.name.includes('post-')) {
+        if (rendition.name.includes('post-')) {
             // to verify post-processing renditions, since their names contain `post-` in the beginning
             rendition.name = rendition.name.substring(5);
 
             const newPath = path.join(rendition.directory, rendition.name);
             await fs.copyFile(rendition.path, newPath);
-        } 
-        
+        }
+
         if (rendition.instructions.name) {
             const newPath = path.join(rendition.directory, rendition.instructions.name);
             await fs.copyFile(rendition.path, newPath);
-        }        
+        }
     } else if (!rendition.shouldEmbedInIOEvent()) {
         await http.upload(rendition);
     }
 }
 
+async function getWatermark(paramsWatermark, inDirectory, disableWatermarkDownload) {
+    // Note: validation has happened in validate.js before
+
+    if (process.env.WORKER_TEST_MODE) {
+        // local file support for `Asset Compute test-worker` unit tests
+        // not supported for clients in production
+        if (!isValidLocalFile(paramsWatermark.watermarkContent, inDirectory)) {
+            throw new Error(`Invalid or missing local file ${paramsWatermark.watermarkContent}`);
+        }
+        return new Watermark(paramsWatermark.watermarkContent, inDirectory);
+
+    } else {
+        const watermark = new Watermark(paramsWatermark, inDirectory);
+
+        if (path.extname(watermark.name).toLowerCase() !== ".png" && !watermark.url.startsWith("data:")) {
```

Comment:
If you use unit tests, disable mockFS completely. 
For file-type usage: https://git.corp.adobe.com/nui/node-file-mimetypes/blob/master/lib/file-type-checker.js#L100. Also make sure you have the correct module, because there are two file-type modules with slightly different names (with and without dash. You want the same as in here: https://git.corp.adobe.com/nui/node-file-mimetypes/blob/master/package.json#L15).

Reply:
Needs a ticket for later